### ISSUE TITLE
ONNX operators needed to be supported

### ISSUE THREAD

**Issue by daquexian** [JDAI-CV/DNNLibrary#17]
This issue is to summarize all operators which needed to be supported. If you want an operator to be supported, please feel free to post it here.

- [x] LRN(#16, supported in #46)
- [x] PRelu(#13, implemented in #23)
- [x] Identify(#13, supported in #24)
- [x] BatchNorm(most batchnorm will be fused into conv so there is no need to support batchnorm in these cases, but a few models(like resnet v2) has a bn->relu->conv sequence, which makes bn unable to be fused into conv.) (Supported in #38 )
- [ ] Resize

**Comment by AmienKhaled** [comment]
Constant operator

**Comment by daquexian** [comment]
PReLU has been supported in #23 

@AmienKhaled I'll implement it soon

**Comment by daquexian** [comment]
@AmienKhaled @yunhaia

In version 0.6.0, identity, prelu and constant op should be supported (please also use the latest onnx2daq)

**Comment by ericdanz** [comment]
Reshape can only be the last layer

**Comment by daquexian** [comment]
@ericdanz reshape layer is somewhat special due to the difference between nhwc(nnapi) and nchw(onnx) layout. I think there is no general, simple way to implement reshape layer, and most reshape layers in an onnx model are indeed generated implicitly. Could you please provide your model so that I can analyze it?

**Comment by ericdanz** [comment]
I can share the onnx file. Quick warning - it's a bit complicated ;) 
https://drive.google.com/file/d/1WdWn12HPr5Ziz69PBk-JyU8YtiEfU2mu/view?usp=sharing

There are a few places reshape is used, in my special batchnorm and shuffle-net like layers.

**Comment by daquexian** [comment]
@ericdanz Thanks for your model. Unfortunately, the reshape here is hard to implement in nnapi. I'll open another issue to track it. However, it may be a long-term work.

**Comment by ericdanz** [comment]
Concat is causing some problems - had to disable the shape checking in shaper.cpp 

Upsample is also not supported yet.  

Made a model that is compatible, about to test it out! Very excited to see more and more ops supported ;)

**Comment by daquexian** [comment]
Thanks for your valuable information!

I will look into it today (It is 7:06 AM, GMT+8 here XD)

Thanks again!

**Comment by mbishop-neurala** [comment]
`Transpose` would be useful when converting caffe models.

**Comment by neginraoof** [comment]
@ericdanz @daquexian Do you happen to have such large pytorch model that could be further optimized after ONNX export?
